test/test.py

```diff
@@ -263,6 +263,25 @@ def tree_diff(path1, path2):
         output = subprocess.check_output([os.path.join(checkout, "tools/tree-diff"), path1, path2])
         return json.loads(output)
 
+    @staticmethod
+    def have_kernel_fs_support(fs: str) -> bool:
+        """Check for Kernel File-System Support
+
+        Check whether the current kernel supports the given file-system. This
+        is done by checking the content of /proc/filesystems.
+        """
+        supported_fs = []
+        with open("/proc/filesystems", "r", encoding="utf-8") as fss:
```

Comment:
While looking at this I noticed that `ext*` filesystems aren't listed on my machine.  Then in a Fedora VM I checked the same and saw `xfs` wasn't there.  From looking around it seems this lists only filesystems that are already loaded, but not necessarily all supported fs modules.  Should maybe look into `/lib/modules/.../kernel/fs` instead. [1]

[1] https://unix.stackexchange.com/a/98680

Reply:
Yeah didn't consider that.  -_-

Reply:
After researching and testing various approaches, I determined that there is basically no way to determine the list of supported filesystems in any environments (container, host) and on any platform. Well, probably except for trying to mount the filesystem, but I didn't want to do this.

I ended up adding a pytest CLI option which will allow us to mark specific filesystems as not supported and tests using it will be skipped. I tested this approach in CentOS Stream MR, where I skip btrfs and all works OK.

The nice thing is also that all filesystems are by default treated as supported, so we won't end up accidentally skipping them in the upstream CI.